src/views/components/AnalyticsGraph/index.vue

```diff
@@ -0,0 +1,312 @@
+      let maxY = d3.max(data, function(d){ return d3.max(d, function(e){return e.amountOfProduct})}) * 1.30;
+
+      let ratio =  height / width;
+
+      let xScale = d3.scaleTime()
+                     .range([0, width])
+                     .domain([minX, maxX]);
+
+      let yScale = d3.scaleLinear()
+                     .range([height, 0])
+                     .domain([minY, maxY]);
+
+     let line = d3.line()
+         .x(function(d) { return xScale(d.datetime); })
+         .y(function(d) { return yScale(d.amountOfProduct); });
+
+     let colors = d3.scaleOrdinal()
+          .domain([0, data.length])
+          .range(d3.schemeCategory20);
+
+    let xAxis = d3.axisBottom(xScale),
+        yAxis = d3.axisLeft(yScale);
+
+    let brush = d3.brush().on("end", brushended),
+        idleTimeout,
+        idleDelay = 350;
+
+    let drag = d3.drag().on('drag', dragged);
+
+    svg.append("g")
+        .attr("class", "brush")
+        .call(brush);
+      // X Axis
+      g.append('g')
+        .attr('class', 'axis--x')
+        .attr('transform', 'translate(0, ' + height + ')')
+        .style("font-weight", "bold")
+        .call(xAxis);
+      // Y Axis
+      g.append('g')
+        .attr('class', 'axis--y')
+        .style("font-weight", "bold")
+        .call(yAxis)
+
+      g.append('defs')
+         .append('clipPath')
+         .attr('id', 'clip')
+         .append('rect')
+           .attr('x', 0)
+           .attr('y', 0)
+           .attr('width', width)
+           .attr('height', height);
+
+      let main = g.append('g')
+         .attr('class', 'main')
+         .attr('clip-path', 'url(#clip)');
+        
+      let legend =g.append('g')
+          .attr('class', 'legend');
+      // iterate over data and generte lines
+      for( let i = 0; i < data.length; i++ ){
+          main.append('path')
+            .datum(data[i])
+            .attr('d', line)
+            .attr('stroke', d => colors(i))
+            .attr('stroke-width', 2)
+            .attr('fill', 'none')
+            .attr('class', 'line');
+
+          main.selectAll('.circle').data(data[i]).enter().append('circle')
+            .attr('cx', function(d) { return xScale(d.datetime); })
+            .attr('cy', function(d) { return yScale(d.amountOfProduct); })
+            .attr('r', 2)
+            .attr('fill', 'white')
+            .attr('stroke', d => colors(i))
+            .attr('stroke-width', 1)
+            .attr('class', 'circles');
+
+          // legend
+          legend = g.append("text")
+            .data(data[i])
+            .attr("transform",
+                "translate(" + (width/ 1.5 + 80 + (90 * i)) + " ," + 
+                              (height / 10 ) + ")")
+            .style("text-anchor", "middle")
+            .style("font-size", "2.2vh") 
+            .attr('stroke', d => colors(i))
+            .text(function(d) {return d.type;});
+      }
+
+      //voronoi
```

Comment:
whats voronoi?

Reply:
its a weird graph that is used by peeps to calculate the closest point to the cursor. Since the line graph is using click and drag to highlight the zoom area I couldn't use on mouseover to highlight the circles so I found this way that calculates its a pain in the ass but works.